### PR TITLE
force merge to publish results against a single run

### PR DESCRIPTION
**Pull request by preetsaimutneja**
- Results are publish against a single run when number of result files is greater than 100. This overrides the options provided by the user in the task.

### CODE REVIEW

File: Tasks/PublishTestResults/publishtestresults.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import * as tl from 'vsts-task-lib/task';
 
+const MERGE_THRESHOLD = 100;
```
Comment (nigurr) [inline Tasks/PublishTestResults/publishtestresults.ts]: add this to task.json info -> publishTestResults option

"If it exceeds these many number of files, we will auto-merge the results

File: Tasks/PublishTestResults/publishtestresults.ts [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import * as tl from 'vsts-task-lib/task';
```
Comment (nigurr) [inline Tasks/PublishTestResults/publishtestresults.ts]: you need to cherry-pick this for m126 also

File: Tasks/PublishTestResults/publishtestresults.ts [hunk 3]
```diff
@@ -21,12 +23,15 @@ if (isNullOrWhitespace(searchFolder)) {
     searchFolder = tl.getVariable('System.DefaultWorkingDirectory');
 }
 
-let matchingTestResultsFiles: string[] = tl.findMatch(searchFolder, testResultsFiles);
+const matchingTestResultsFiles: string[] = tl.findMatch(searchFolder, testResultsFiles);
+const forceMerge = matchingTestResultsFiles && matchingTestResultsFiles.length > MERGE_THRESHOLD;
+tl.debug(`forceMerge is set to ${forceMerge} based on the the number of Test Results files.`);
```
Comment (nigurr) [inline Tasks/PublishTestResults/publishtestresults.ts]: this should be warning as we are forcefully doing it - localization

File: Tasks/PublishTestResults/Strings/resources.resjson/en-US/resources.resjson [hunk 4]
```diff
@@ -12,13 +12,14 @@
   "loc.input.label.searchFolder": "Search folder",
   "loc.input.help.searchFolder": "Folder to search for the test result files. Defaults to $(System.DefaultWorkingDirectory).",
   "loc.input.label.mergeTestResults": "Merge test results",
-  "loc.input.help.mergeTestResults": "To merge results from all test results files to one test run in VSTS/TFS, check this option. To create a test run for each test results file, uncheck this option.",
+  "loc.input.help.mergeTestResults": "A test run is created for each results file. Check this option to merge results into a single test run. To optimize for better performance, results will be merged into a single run if there are more than 100 result files, irrespective of this option.",
```
Comment (preetsaimutneja) [inline Tasks/PublishTestResults/Strings/resources.resjson/en-US/resources.resjson]: suggested by @ManojBableshwar

File: Tasks/PublishTestResults/publishtestresults.ts (hunk 1 above)
Comment (nigurr) [reply Tasks/PublishTestResults/publishtestresults.ts]: we should ideally write one service to get all test related configuration and consume it here. approving it for now. but to do create WIT

**Comment by stephenmichaelf** [comment]
@preetsaimutneja Is this going in or can we close it?

**Pull request closed without merging**